### PR TITLE
CLI workspaces

### PR DESCRIPTION
**Pull request by CrossR**
This starts to adapt some of the logic from #1707 to do with swapping workspaces.

One of the issues I'm hitting is not quite knowing what logic is best to implement. Currently, I've implemented the logic from the issue, without the `"workspace.autoDetectWorkspace"` logic.

That is, in the current form of the PR, the logic is as follows, in order of priority:
- CLI : Use first given folder, use first given file, use location `oni` was called from.
- Non-CLI (unchanged): Use first given folder, use first given file.

Neither of these can be changed with a config option, is that something we'd want?

Things I'd potentially think are useful would be the ability to never use the location `oni` was called from, and the ability to always/never use the detection logic.

I'll tag @badosu since I know he had some thoughts about this over in #2163.

Anyone any ideas?

### CODE REVIEW

**Comment by bryphe** [comment]
> Neither of these can be changed with a config option, is that something we'd want?

The behavior sounds reasonable to me (and in-line with expections of CLI behavior for similiar apps). So I'd lean towards not adding configuration for now; if there ends up being a need for it, we can add it later. 

> Things I'd potentially think are useful would be the ability to never use the location oni was called from, and the ability to always/never use the detection logic.

There could potentially be the workaround today of adding logic in the plugin load to force a specific workspace, too - so it may be that we don't even need a configuration setting to accomodate these scenarios. For example, I could force a workspace via `oni.workspace.changeDirectory("/my/permanent/folder")` in the `activate` method of my plugin. 

The plugin also has access to the arguments via `process.argv` (although this is serendipity, as we host the Oni plugins in the renderer process) - so it could do custom processing. I think it would make sense to make the CLI arguments available somewhere on our API too, though.

**Comment by CrossR** [comment]
Hoping to get this updated, tested and then merged, since there wasn't any extra feedback that the other stuff was needed.

I'll also have a look into the dev setup for all this, since my scripts are all aimed at the production build, but I'm now realising they should also work for the users who run master but without building fully.

**Comment by CrossR** [comment]
Okay, I've finally been able to tidy this up.

I'll need to stick up a PR for oni-api, and check the functionality a bit more, but this adds what the PR initial message says + support for using NETRW. (Ie passing a folder in and it will open netrw on that folder).

File: browser/src/Editor/NeovimEditor/NeovimEditor.tsx [hunk 1]
```diff
@@ -1045,9 +1097,21 @@ export class NeovimEditor extends Editor implements Oni.Editor {
             await this.setColorSchemeFromTheme(this._themeManager.activeTheme)
         }
 
+        let openWelcome = true
```
[the remark this answers is not in the document]
Comment (CrossR) [reply browser/src/Editor/NeovimEditor/NeovimEditor.tsx]: Sorted this logic now!

**Comment by bryphe** [comment]
> Okay, I've finally been able to tidy this up.

> I'll need to stick up a PR for oni-api, and check the functionality a bit more, but this adds what the PR initial message says + support for using NETRW. (Ie passing a folder in and it will open netrw on that folder).

Thanks for all the work on this, @CrossR ! Excited to see it go in 👍

**Pull request closed without merging**